lib/Runtime/Types/Type.h

```diff
@@ -62,6 +63,14 @@ namespace Js
         inline BOOL SkipsPrototype() const { return (this->flags & TypeFlagMask_SkipsPrototype) != 0 ; }
         inline BOOL CanHaveInterceptors() const { return (this->flags & TypeFlagMask_CanHaveInterceptors) != 0; }
         inline BOOL IsFalsy() const { return flags & TypeFlagMask_IsFalsy; }
+        inline BOOL HasBeenCached() const { return flags & TypeFlagMask_HasBeenCached; }
+        inline void SetHasBeenCached(const bool hasBeenCached)
+        {
+            // Once set, this flag should never be reset.
```

Comment:
> is an optimization, not a functional change

Sure, I get that. But your change is optimizing that very case. Correct me if I am wrong, but won't it be more complete to optimize for cases where types are invalidated (which happens a lot all the time). 

---
In reply to: [78828446](https://github.com/Microsoft/ChakraCore/pull/1542#discussion_r78828446) [](ancestors = 78828446,78818005,78816821,78790053)